src/libraries/System.Linq.Queryable/src/System/Linq/CachedReflection.cs

```diff
@@ -14,36 +14,42 @@ internal static class CachedReflectionInfo
     {
         private static MethodInfo? s_Aggregate_TSource_2;
 
+        [RequiresDynamicCode("Calls System.Reflection.MethodInfo.MakeGenericMethod(params Type[])")]
```

Comment:
This message might not be very user friendly in terms to what the problem is and how to work around it

Reply:
This is an internal class so I don't think we need to be more specific.